File: src/main/java/no/nav/syfo/mock/ArbeidsforholdMock.java
package no.nav.syfo.mock;

import no.nav.tjeneste.virksomhet.arbeidsforhold.v3.binding.*;
import no.nav.tjeneste.virksomhet.arbeidsforhold.v3.informasjon.arbeidsforhold.Arbeidsforhold;
import no.nav.tjeneste.virksomhet.arbeidsforhold.v3.informasjon.arbeidsforhold.Organisasjon;
import no.nav.tjeneste.virksomhet.arbeidsforhold.v3.meldinger.*;

public class ArbeidsforholdMock implements ArbeidsforholdV3 {
    public FinnArbeidsforholdPrArbeidsgiverResponse finnArbeidsforholdPrArbeidsgiver(FinnArbeidsforholdPrArbeidsgiverRequest parameters)
            throws FinnArbeidsforholdPrArbeidsgiverForMangeForekomster, FinnArbeidsforholdPrArbeidsgiverSikkerhetsbegrensning, FinnArbeidsforholdPrArbeidsgiverUgyldigInput {
        throw new RuntimeException("Ikke implementert i mock. Se ArbeidsforholdMock");
    }

    public FinnArbeidsforholdPrArbeidstakerResponse finnArbeidsforholdPrArbeidstaker(FinnArbeidsforholdPrArbeidstakerRequest parameters)
            throws FinnArbeidsforholdPrArbeidstakerSikkerhetsbegrensning, FinnArbeidsforholdPrArbeidstakerUgyldigInput {
        Arbeidsforhold arbeidsforhold = new Arbeidsforhold();
        Organisasjon arbeidsgiver = new Organisasjon();
        arbeidsgiver.setOrgnummer("111222333");
        arbeidsforhold.setArbeidsgiver(arbeidsgiver);
        FinnArbeidsforholdPrArbeidstakerResponse response = new FinnArbeidsforholdPrArbeidstakerResponse();
        response.getArbeidsforhold().add(arbeidsforhold);
        return response;
    }

    public HentArbeidsforholdHistorikkResponse hentArbeidsforholdHistorikk(HentArbeidsforholdHistorikkRequest parameters)
            throws HentArbeidsforholdHistorikkArbeidsforholdIkkeFunnet, HentArbeidsforholdHistorikkSikkerhetsbegrensning {
        throw new RuntimeException("Ikke implementert i mock. Se ArbeidsforholdMock");
    }

    public FinnArbeidstakerePrArbeidsgiverResponse finnArbeidstakerePrArbeidsgiver(FinnArbeidstakerePrArbeidsgiverRequest parameters)
            throws FinnArbeidstakerePrArbeidsgiverSikkerhetsbegrensning, FinnArbeidstakerePrArbeidsgiverUgyldigInput {
        throw new RuntimeException("Ikke implementert i mock. Se ArbeidsforholdMock");
    }

    public void ping() {
    }
}

File: src/main/kotlin/no/nav/syfo/mock/ArbeidsforholdMock.kt
package no.nav.syfo.mock

import no.nav.syfo.config.AAregConfig
import no.nav.tjeneste.virksomhet.arbeidsforhold.v3.binding.*
import no.nav.tjeneste.virksomhet.arbeidsforhold.v3.informasjon.arbeidsforhold.Arbeidsforhold
import no.nav.tjeneste.virksomhet.arbeidsforhold.v3.informasjon.arbeidsforhold.Organisasjon
import no.nav.tjeneste.virksomhet.arbeidsforhold.v3.meldinger.*
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.stereotype.Service

@Service
@ConditionalOnProperty(value = [AAregConfig.MOCK_KEY], havingValue = "true")
class ArbeidsforholdMock : ArbeidsforholdV3 {
    @Throws(FinnArbeidsforholdPrArbeidsgiverForMangeForekomster::class, FinnArbeidsforholdPrArbeidsgiverSikkerhetsbegrensning::class, FinnArbeidsforholdPrArbeidsgiverUgyldigInput::class)
    override fun finnArbeidsforholdPrArbeidsgiver(parameters: FinnArbeidsforholdPrArbeidsgiverRequest): FinnArbeidsforholdPrArbeidsgiverResponse {
        throw RuntimeException("Ikke implementert i mock. Se ArbeidsforholdMock")
    }

    @Throws(FinnArbeidsforholdPrArbeidstakerSikkerhetsbegrensning::class, FinnArbeidsforholdPrArbeidstakerUgyldigInput::class)
    override fun finnArbeidsforholdPrArbeidstaker(parameters: FinnArbeidsforholdPrArbeidstakerRequest): FinnArbeidsforholdPrArbeidstakerResponse {
        val arbeidsforhold = Arbeidsforhold()
        val arbeidsgiver = Organisasjon()
        arbeidsgiver.orgnummer = "111222333"
        arbeidsforhold.arbeidsgiver = arbeidsgiver
        val response = FinnArbeidsforholdPrArbeidstakerResponse()
        response.arbeidsforhold.add(arbeidsforhold)
        return response
    }

    @Throws(HentArbeidsforholdHistorikkArbeidsforholdIkkeFunnet::class, HentArbeidsforholdHistorikkSikkerhetsbegrensning::class)
    override fun hentArbeidsforholdHistorikk(parameters: HentArbeidsforholdHistorikkRequest): HentArbeidsforholdHistorikkResponse {
        throw RuntimeException("Ikke implementert i mock. Se ArbeidsforholdMock")
    }

    @Throws(FinnArbeidstakerePrArbeidsgiverSikkerhetsbegrensning::class, FinnArbeidstakerePrArbeidsgiverUgyldigInput::class)
    override fun finnArbeidstakerePrArbeidsgiver(parameters: FinnArbeidstakerePrArbeidsgiverRequest): FinnArbeidstakerePrArbeidsgiverResponse {
        throw RuntimeException("Ikke implementert i mock. Se ArbeidsforholdMock")
    }

    override fun ping() {}
}
